besu/src/main/java/org/hyperledger/besu/services/BesuPluginContextImpl.java

```diff
@@ -142,10 +142,11 @@ public void registerPlugins(final PluginConfiguration config) {
           matchAndValidateRequestedPlugins(requestedPlugins, detectedPlugins);
 
       registerPlugins(registeringPlugins);
-    } else {
-      // If no plugins were specified, register all detected plugins
+    } else if (config.isPluginAutoEnabled()) {
```

Comment:
So, if pluginAutoEnable is true, but we also specify some requested plugins, it will only load the requested plugins? Could be a bit confusing if we don't explicitly explain this behavior in documentation.

Reply:
That is the flow:

```mermaid
graph TD
    B{--plugins specified?} -->|Yes| C[Load specified plugins]
    B -->|No| D{--Xplugin-auto-enabled?}
    D -->|Yes| E[Auto-register plugins]
    D -->|No| F[Do not register plugins]

Reply:
If --plugins are specified and --Xplugin-auto-enabled are specified then I think there should be an error as you have specified plugins but disabled the mechanism to do so.

Reply:
Do you prefer the option to disable plugins at all?